.github/workflows/metric.yml

```diff
@@ -0,0 +1,71 @@
+name: Metric and Commit Results
+
+on:
+  workflow_dispatch: # Allows manual triggering of the workflow
+  push:
+    branches:
+      - main
+    paths-ignore:
+      - '**/*.md'
+
+jobs:
+  compile-time:
+    name: Compile Time
+    runs-on: ubuntu-latest
+    # if: ${{ github.ref_name == 'main' }}
+    outputs:
+      time: ${{ steps.run.outputs.COMPILE_TIME }}
+    steps:
+      - uses: taiki-e/checkout-action@v1
+      - name: Checkout repo
+        uses: actions/checkout@v4
+      - uses: Boshen/setup-rust@main
+      - run: |
+          cargo fetch
+          cargo clean # build from scratch using local system cache to avoid download from crates.io
+      - id: run
+        run: |
+          /usr/bin/time -f "%e" -o output cargo build --release
+          cat output
+          COMPILE_TIME=$(cat output)
+          echo $COMPILE_TIME
+          echo "COMPILE_TIME=$COMPILE_TIME" >> $GITHUB_OUTPUT
+
+  metric:
+    name: Metric
+    needs: [compile-time]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout rolldown metric repository
+        uses: actions/checkout@v4
+        with:
+          repository: 'rolldown/metric'
+          ref: 'main'
+          token: ${{ secrets.METRIC_SECRET_KEY }}
+
+      - name: Setup Pnpm
+        uses: pnpm/action-setup@v2
+        with:
+          version: 8.10.5
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20' # Specify the Node.js version you need
+          cache: 'pnpm'
+
+      - name: Install dependencies and push metric
+        run: |
+          pnpm install -r
```

Comment:
It seems like the scripts don't need any dependencies installed? Otherwise you're downloading all the bundlers, which is useless.

Reply:
need install devDependency to execute node script, `node ./scripts/compile-time.mjs`